controller/settings.go

```diff
@@ -47,3 +47,35 @@ func (o OnConnectSettingsHandler) RouterConnected(r *network.Router) {
 		}).Info("no on connect settings to send")
 	}
 }
+
+type OnConnectCtrlAddressesUpdateHandler struct {
+	callback func() []string
+}
+
+func (o *OnConnectCtrlAddressesUpdateHandler) RouterDisconnected(r *network.Router) {
+	//do nothing, satisfy interface
+}
+
+func (o OnConnectCtrlAddressesUpdateHandler) RouterConnected(r *network.Router) {
+	pfxlog.Logger().Info("Router connected... syncing crtl addresses")
+	data := o.callback()
+	pfxlog.Logger().Info(data)
+	updMsg := &ctrl_pb.UpdateCtrlAddresses{
+		Addresses: data,
+	}
+
```

Comment:
If you add a GetContentType for `UpdateCtrlAddresses` in ctrl_pb/impl.go, you can do 

```
protobufs.MarshalTyped(updMsg).Send(r.Control)
```